### DISCUSSION TITLE
How can I use cameraundistort with rtsp stream?

### DISCUSSION THREAD

**Discussion by rtcnerd** [awslabs/amazon-kinesis-video-streams-producer-sdk-cpp#1005] [Q&A] [answered]
I would like to pull a rtsp stream from an IP camera. The image from the ipcamera is fisheye. I would like to use cameraundistort with the KVS pipeline. How can I achieve that?

Thanks.

**Comment by sirknightj** [comment, marked as answer]
You're probably looking for the gstreamer element `dewarp`, as the documentation for cameraundistort mentions that it performs camera calibration.

The use case described sounds like we want to take the video from the camera, transform it, then send it to KVS. 

Suppose we use the sample pipeline from [here](https://docs.aws.amazon.com/kinesisvideostreams/latest/dg/examples-gstreamer-plugin.html#examples-gstreamer-plugin-launch-ex1). We need to inject a `dewarp` element in it. 

According to its documentation, [dewarp](https://gstreamer.freedesktop.org/documentation/opencv/dewarp.html?gi-language=c) only accepts RGBA in and RGBA out. So, we will need to convert it from: h264 (camera) -> RGBA (dewarp) ->  h264 -> kvssink 

Something along these lines might do it.
```
gst-launch-1.0 -v rtspsrc location="x" short-header=TRUE ! rtph264depay ! decodebin ! videoconvert ! video/x-raw,format=RGBA ! dewarp outer-radius=0.35 inner-radius=0.1 ! videoconvert ! x264enc ! kvssink stream-name="x" storage-size=128
```

( `sudo apt-get install gstreamer1.0-opencv` to get the dewarp element)
